File: api/ServerManager.Rest/Dto/UpdateServerRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ServerManager.Rest.Dto
{
    public class UpdateServerRequest
    {
        public string NewName { get; set; }
        public string Version { get; set; }
        public string Description { get; set; }
        public ServerPropertyList NewProperties { get; set; }
    }
}

File: ui/management-console/src/models/UpdateServerRequest.ts
import { ServerProperty } from "./ServerProperty";

export class UpdateServerRequest {
	public constructor() {
		this.newName = "";
		this.version = "";
		this.description = "";
		this.newProperties = [];
	}

	newName: string;
	version: string;
	description: string;
	newProperties: ServerProperty[];
}
